tortoise/backends/asyncpg/client.py

```diff
@@ -39,6 +39,26 @@ def __init__(self, user, password, database, host, port, *args, **kwargs):
         )
 
     async def create_connection(self):
+        if self.create_db:
+            single_connection = self.single_connection
+            self.single_connection = True
+            self._connection = await asyncpg.connect(self.DSN_TEMPLATE.format(
+                user=self.user,
+                password=self.password,
+                host=self.host,
+                port=self.port,
+                database=''
+            ))
+            try:
+                await self.execute_script('DROP DATABASE {}'.format(self.database))
+            except asyncpg.InvalidCatalogNameError:
+                pass
+            await self.execute_script(
+                'CREATE DATABASE {} OWNER {}'.format(self.database, self.user)
+            )
+            await self._connection.close()
+            self.single_connection = single_connection
+
```

Comment:
I really don't like idea of possible wiping out current db just by passing a parameter to connection method, it looks really unsafe to me. I think creating schema must be done in some other place, may be you can expand `schema_generator`, something like `client.schema_generator.create_db()`

Reply:
Hmm, The reason I added both create and delete parameters is for the case where you Don't want the test runner to destroy the self made db.

This is really just behaviour for the test runner, but my initial attempt to put it outside the db drivers was super super hacky.
We can at worst make it so that a DB can only be deleted if it was created by the same connection?

Please propose a better way of doing this?

Reply:
I think you can just make explicit methods for `db_create` and `db_delete` on DbClient and use them. In that case using one of this methods will be fully user own responsibility and it will be hard to misuse such methods. It also seems quite easy to implement code-wise